src/main/java/org/spongepowered/api/advancement/criteria/trigger/TriggerTypes.java

```diff
@@ -0,0 +1,90 @@
+
+    public static final TriggerType BRED_ANIMALS = DummyObjectProvider.createFor(TriggerType.class, "BRED_ANIMALS");
+
+    public static final TriggerType BREWED_POTION = DummyObjectProvider.createFor(TriggerType.class, "BREWED_POTION");
+
+    public static final TriggerType CONSTRUCT_BEACON = DummyObjectProvider.createFor(TriggerType.class, "CONSTRUCT_BEACON");
+
+    public static final TriggerType CONSUME_ITEM = DummyObjectProvider.createFor(TriggerType.class, "CONSUME_ITEM");
+
+    public static final TriggerType CURSE_ZOMBIE_VILLAGER = DummyObjectProvider.createFor(TriggerType.class, "CONSUME_ITEM");
+
+    public static final TriggerType ENCHANTED_ITEM = DummyObjectProvider.createFor(TriggerType.class, "ENCHANTED_ITEM");
+
+    public static final TriggerType ENTER_BLOCK = DummyObjectProvider.createFor(TriggerType.class, "ENTER_BLOCK");
+
+    public static final TriggerType ENTITY_HURT_PLAYER = DummyObjectProvider.createFor(TriggerType.class, "ENTITY_HURT_PLAYER");
+
+    public static final TriggerType ENTITY_KILLED_PLAYER = DummyObjectProvider.createFor(TriggerType.class, "ENTITY_KILLED_PLAYER");
+
+    public static final TriggerType IMPOSSIBLE = DummyObjectProvider.createFor(TriggerType.class, "IMPOSSIBLE");
+
+    public static final TriggerType INVENTORY_CHANGED = DummyObjectProvider.createFor(TriggerType.class, "INVENTORY_CHANGED");
+
+    public static final TriggerType ITEM_DURABILITY_CHANGED = DummyObjectProvider.createFor(TriggerType.class, "ITEM_DURABILITY_CHANGED");
+
+    public static final TriggerType LOCATION = DummyObjectProvider.createFor(TriggerType.class, "LOCATION");
+
+    public static final TriggerType PLACED_BLOCK = DummyObjectProvider.createFor(TriggerType.class, "PLACED_BLOCK");
+
+    public static final TriggerType PLAYER_HURT_ENTITY = DummyObjectProvider.createFor(TriggerType.class, "PLAYER_HURT_ENTITY");
+
+    public static final TriggerType PLAYER_KILLED_ENTITY = DummyObjectProvider.createFor(TriggerType.class, "PLAYER_KILLED_ENTITY");
+
+    public static final TriggerType RECIPE_UNLOCKED = DummyObjectProvider.createFor(TriggerType.class, "RECIPE_UNLOCKED");
+
+    public static final TriggerType SLEPT_IN_BED = DummyObjectProvider.createFor(TriggerType.class, "SLEPT_IN_BED");
+
+    public static final TriggerType SUMMONED_ENTITY = DummyObjectProvider.createFor(TriggerType.class, "SUMMONED_ENTITY");
+
+    public static final TriggerType TAME_ANIMAL = DummyObjectProvider.createFor(TriggerType.class, "TAME_ANIMAL");
+
+    public static final TriggerType TICK = DummyObjectProvider.createFor(TriggerType.class, "TICK");
+
+    public static final TriggerType USED_ENDER_EYE = DummyObjectProvider.createFor(TriggerType.class, "USED_ENDER_EYE");
+
+    public static final TriggerType VILLAGER_TRADE = DummyObjectProvider.createFor(TriggerType.class, "VILLAGER_TRADE");
+
+    public static final TriggerType CHANGED_DIMENSION = DummyObjectProvider.createFor(TriggerType.class, "CHANGED_DIMENSION");
```

Comment:
This should be before CONSTRUCT_BEACON

Reply:
All of these string labels "VILLAGER_TRADE" should be moved into a class holding static final strings.  I am loathe to see embedded literal strings in code as much as possible.

Reply:
just run sortfield once more and it will be fine.

Reply:
@sibomots The strings here are used only for debugging. the dummy objects are nearly useless and as far as I'm aware don't even necessarily need correct names outside of logging in the case that something goes terribly wrong.

Yes it's a gross hack, but it's a gross hack that was decided on by the core team a long while ago, and isn't the subject of this PR outside correctly the incorrect debug text.